### DISCUSSION TITLE
wireguard arm

### DISCUSSION THREAD

**Discussion by gxf77** [Freetz-NG/freetz-ng#464] [Allgemein] [answered]
Hallo 
ich bebommen beim erstellen bei der 7530/7530 Build error :-( 
könntest du dir das mal bitte anschauen 
getestet auf Debian 10 , 11 Ubuntu und Mint 

> make[2]: Verzeichnis „/home/freetz/freetz-ng/source/kernel/ref-ipq40xx-7530_07.25/linux-4.4“ wird verlassen
make[1]: Verzeichnis „/home/freetz/freetz-ng/source/kernel/ref-ipq40xx-7530_07.25/linux-4.4“ wird verlassen
touch source/kernel/ref-ipq40xx-7530_07.25/.modules-ipq40xx
rm -f -r kernel/modules-ipq40xx-7530_07.25/lib
mkdir -p kernel/modules-ipq40xx-7530_07.25
tools/tar-gnu -cf - -C source/kernel/ref-ipq40xx-7530_07.25/lib/modules/4.4.60/kernel --exclude=.svn --exclude=.git --exclude=.gitignore --exclude=.build-prereq-checked --exclude=.unpacked --exclude=.configured --exclude=.compiled --exclude=.installed . | tools/tar-gnu -xf - -C kernel/modules-ipq40xx-7530_07.25;
touch kernel/modules-ipq40xx-7530_07.25/.modules-ipq40xx
done.
cmd() { PATH="/home/freetz/freetz-ng/toolchain/build/arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/arm-linux-uclibcgnueabi/bin:/home/freetz/freetz-ng/toolchain/build/arm_gcc-5.5.0/arm-unknown-linux-gnueabi/bin:/home/freetz/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin" LD_RUN_PATH="/usr/lib/freetz" FREETZ_LIBRARY_DIR="/usr/lib/freetz" make -j2  "$@"  || { printf "\n\\033[33m%s\\033[m\n" "ERROR: Build failed.";  exit 1; } }; 	if [ -e source/.echo_item_start -a ! -e source/.echo_item_build ]; then echo -ne "\e[48;5;56mbuilding\e[49m ... "; touch source/.echo_item_build; fi; cmd -C source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src \
	SUBDIRS="/home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src" \
	KERNELDIR="/home/freetz/freetz-ng/source/kernel/ref-ipq40xx-7530_07.25/linux-4.4" \
	ARCH="arm" \
	CROSS_COMPILE="arm-unknown-linux-gnueabi-"
make[1]: Verzeichnis „/home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src“ wird betreten
  CC [M]  /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/main.o
  CC [M]  /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/noise.o
In file included from <command-line>:0:0:
/home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/compat/compat.h:690:21: error: redefinition of 'skb_put_data'
 static inline void *skb_put_data(struct sk_buff *skb, const void *data, unsigned int len)
                     ^
In file included from <command-line>:0:0:
/home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/compat/compat.h:690:21: error: redefinition of 'skb_put_data'
 static inline void *skb_put_data(struct sk_buff *skb, const void *data, unsigned int len)
                     ^
In file included from include/linux/ip.h:20:0,
                 from include/linux/if_tunnel.h:4,
                 from include/net/ip_tunnels.h:4,
                 from include/net/udp_tunnel.h:4,
                 from /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/compat/udp_tunnel/udp_tunnel_partial_compat.h:143,
                 from /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/compat/compat.h:73,
                 from <command-line>:0:
include/linux/skbuff.h:1885:21: note: previous definition of 'skb_put_data' was here
 static inline void *skb_put_data(struct sk_buff *skb, const void *data,
                     ^
In file included from include/linux/ip.h:20:0,
                 from include/linux/if_tunnel.h:4,
                 from include/net/ip_tunnels.h:4,
                 from include/net/udp_tunnel.h:4,
                 from /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/compat/udp_tunnel/udp_tunnel_partial_compat.h:143,
                 from /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/compat/compat.h:73,
                 from <command-line>:0:
include/linux/skbuff.h:1885:21: note: previous definition of 'skb_put_data' was here
 static inline void *skb_put_data(struct sk_buff *skb, const void *data,
                     ^
make[4]: *** [scripts/Makefile.build:258: /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/main.o] Fehler 1
make[4]: *** Auf noch nicht beendete Prozesse wird gewartet …
make[4]: *** [scripts/Makefile.build:258: /home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/noise.o] Fehler 1
make[3]: *** [Makefile:1408: _module_/home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src] Fehler 2
make[2]: *** [GNUmakefile:109: modules] Fehler 2
make[1]: *** [Makefile:26: module] Fehler 2
make[1]: Verzeichnis „/home/freetz/freetz-ng/source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src“ wird verlassen

ERROR: Build failed.
make: *** [make/wireguard-linux-compat/wireguard-linux-compat.mk:24: source/target-arm_gcc-5.5.0_uClibc-1.0.40-nptl_kernel-4.4/wireguard-linux-compat-1.0.20211208/src/wireguard.ko] Fehler 1

**Comment by gxf77** [comment, marked as answer]
Vielen Dank für die schnelle Arbeit 
Aktualisiere 6ea279b3e..ac4c7214a
Feedback  -->  arm 7520/7530  build erfolgreich
kann geschlossen werden

**Comment by fda77** [comment]
Schliessen kann man nur ein "issue", einer "Diskussion" auf Github kann man nur als beantwortet markieren. Auch werden Commits die mit "refs" genannt werden hier nicht angezeigt
